🐙 source-github: run up-to-date pipeline [2025-06-07]

airbyteio opened the pull request:
# Update source-github

This PR was autogenerated by running `airbyte-ci connectors --name=source-github up_to_date --pull`

We've set the `auto-merge` label on it, so it will be automatically merged if the CI pipelines pass.
If you don't want to merge it automatically, please remove the `auto-merge` label.
Please reach out to the Airbyte Connector Tooling team if you have any questions or concerns.


## Operations

- Upgrade the base image to the latest version in metadata.yaml: Skipped

- Update versions of libraries in poetry.: Successful

- PATCH bump source-github version to 1.8.27: Successful

- Build source-github docker image for platform(s) linux/amd64, linux/arm64: Successful

- Get dependency updates: Successful

- Create or update pull request on Airbyte repository: Successful

- Add changelog entry: Successful




## Dependency updates

We use [`syft`](https://github.com/anchore/syft) to generate a SBOM for the latest connector version and the one from the PR.
It allows us to spot the dependencies that have been updated at all levels and for all types of dependencies (system, python, java etc.).
Here are the dependencies that have been updated compared to `airbyte/source-github:latest`.
Keep in mind that `:latest` does not always match the connector code on the main branch. 
It is the latest released connector image when the head commit of this branch was created.

| Type | Name | State | Previous Version | New Version |
|------|------|-------|-------------|------------------|
| python | Jinja2 | updated | 3.1.5 | **3.1.6** |
| python | anyio | updated | 4.8.0 | **4.9.0** |
| python | attrs | updated | 25.1.0 | **25.3.0** |
| python | cachetools | updated | 5.5.2 | **6.0.0** |
| python | cattrs | updated | 24.1.2 | **25.1.1** |
| python | certifi | updated | 2025.1.31 | **2025.4.26** |
| python | charset-normalizer | updated | 3.4.1 | **3.4.2** |
| python | click | updated | 8.1.8 | **8.2.1** |
| python | h11 | updated | 0.14.0 | **0.16.0** |
| python | httpcore | updated | 1.0.7 | **1.0.9** |
| python | joblib | updated | 1.4.2 | **1.5.1** |
| python | orjson | updated | 3.10.15 | **3.10.18** |
| python | platformdirs | updated | 4.3.6 | **4.3.8** |
| python | pydantic | updated | 2.10.6 | **2.11.5** |
| python | pydantic_core | updated | 2.27.2 | **2.33.2** |
| python | setuptools | updated | 75.8.0 | **80.9.0** |
| python | typing_extensions | updated | 4.12.2 | **4.14.0** |
| python | url-normalize | updated | 1.4.3 | **2.2.1** |
| python | urllib3 | updated | 2.3.0 | **2.4.0** |
| python | typing-inspection | added | not present | **0.4.1** |

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=54931) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=54931) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=54931) it.</sub>

airbyteio committed:
Update versions of libraries in poetry.
PATCH bump source-github version to 1.8.27
Build source-github docker image for platform(s) linux/amd64, linux/arm64
Get dependency updates
Create or update pull request on Airbyte repository
Add changelog entry

dbgold17 committed:
Merge branch 'master' into up-to-date/source-github

dbgold17 committed:
override poe unit test command

dbgold17 commented:
I determined that there appears to be some sort of incompatibility between two request mocking packages used by the unit tests for source-github (`requests_mock` and `responses`), which led to the failures in this PR. This issue was introduced by upgrading `pytest` to version 8 in [this PR](https://github.com/airbytehq/airbyte/issues/54707).

The workaround for now is to run two sets of integration tests in separate calls to pytest.

aaronsteers commented on airbyte-integrations/connectors/source-github/pyproject.toml:
```diff
@@ -37,3 +37,12 @@ include = [
     "${POE_GIT_DIR}/poe-tasks/poetry-connector-tasks.toml",
 ]
+
+[tool.poe.tasks]
+# This overrides the default task for running unit tests specifically or the source-github connector.
+# This is a workaround to an issue that arises when running all of the unit tests in one execution of pytest 
+# due to incompatibilities between request mocking libraries requests-mock and responses.
+test-unit-tests.shell = '''
+poetry run pytest --junitxml=build/test-results/pytest-unit-tests-junit.xml --ignore=unit_tests/integration/ unit_tests
+poetry run pytest --junitxml=build/test-results/pytest-unit-integration-tests-junit.xml unit_tests/integration/
+'''
```
👀

dbgold17 replied:
seems like it does. 82 is the total number of all unit tests

aaronsteers replied:
@dbgold17 - Totally! The junit plugin just grabs anything that matches `build/test-results/*.xml`. If anything, this gives more granularity when reported, since I think the different files will be grouped individually. Definitely should not be a problem.

aaronsteers reviewed (approved):
Nice work on this, @dbgold17!

I want to learn more about the testing issue and the need for the override, but we can tackle that offline, probably in a long-running thread...

No reason to block on the test nuances, so I'm approving for my part. 🚀 ✅

dbgold17 committed:
fix typo

ChristoGrab reviewed (approved):
Nice, clever workaround! :shipit:

airbyteio committed:
Update versions of libraries in poetry.
PATCH bump source-github version to 1.8.27
Build source-github docker image for platform(s) linux/amd64, linux/arm64
Get dependency updates
Create or update pull request on Airbyte repository
Add changelog entry

dbgold17 committed:
Merge remote-tracking branch 'refs/remotes/origin/up-to-date/source-github' into up-to-date/source-github

dbgold17 commented:
One CAT test failing with `AssertionError: All streams should return some records, streams without records: {'project_cards', 'project_columns', 'projects'}` however the same failure is present on master so I don't think it has anything to do with these changes. I created an issue [here](https://github.com/airbytehq/airbyte-internal-issues/issues/13284) to fix these failing tests.
